### PR TITLE
Angular2 project added

### LINKED ISSUE #46: Fix problems with JS
@FreexD reports some problem with Angular2. The task is to resolve those problems and make solution work.

### PR DESCRIPTION
**Pull request by gmrukwa**
**Setup:** A script `scripts\InstallNodeDev.ps1` was prepared to simplyify installation process for all of you. It installs you Node.js in the version we use along with two global tools (*npm* and [*angular-cli*](https://github.com/angular/angular-cli#generating-and-serving-an-angular-project-via-a-development-server)). First is a package manager necessary in each case, second allows to manage project components and running.

**What's been done:** A skeleton of TypeScript project for Angular2 has been added. It consists of a *Hello World!*-like page. Unit tests are introduced, as well as end-to-end tests (something similar to Selenium, but designed and cooperating much better with Angular2 specifically). Moreover, all the JS tests are causing AppVeyor build to fail in the case tests do not pass.

Resolves #46 

**Note:** No license headers at the moment, but it's all auto-generated code, which is just a base for *any* further development.

### CODE REVIEW

**Comment by FreexD** [comment]
LGTM

File: src/Spectre.Angular2Client.Tests/Spectre.Angular2Client.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,100 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
```
Comment (Daroslaw) [inline src/Spectre.Angular2Client.Tests/Spectre.Angular2Client.Tests.csproj]: x64?

File: src/Spectre.Angular2Client.Tests/Spectre.Angular2Client.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    <ProjectTypeGuids>{3AC096D0-A1C2-E12C-1390-A8335801FDAB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
+    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">10.0</VisualStudioVersion>
+    <VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>
+    <ReferencePath>$(ProgramFiles)\Common Files\microsoft shared\VSTT\$(VisualStudioVersion)\UITestExtensionPackages</ReferencePath>
+    <IsCodedUITest>False</IsCodedUITest>
+    <TestProjectType>UnitTest</TestProjectType>
+    <NuGetPackageImportStamp>
+    </NuGetPackageImportStamp>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
```
Comment (Daroslaw) [inline src/Spectre.Angular2Client.Tests/Spectre.Angular2Client.Tests.csproj]: x64?

File: src/Spectre.Angular2Client.Tests/Spectre.Angular2Client.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    <DebugSymbols>true</DebugSymbols>
+    <DebugType>full</DebugType>
+    <Optimize>false</Optimize>
+    <OutputPath>bin\Debug\</OutputPath>
+    <DefineConstants>DEBUG;TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
```
Comment (Daroslaw) [inline src/Spectre.Angular2Client.Tests/Spectre.Angular2Client.Tests.csproj]: x64?

File: scripts/InstallNodeDev.ps1 [hunk 4]
```diff
@@ -0,0 +1,13 @@
+if (!([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator")) { Start-Process powershell.exe "-NoProfile -ExecutionPolicy Bypass -File `"$PSCommandPath`"" -Verb RunAs; exit }
+
+Invoke-WebRequest https://nodejs.org/dist/v6.10.0/node-v6.10.0-x64.msi -OutFile node.msi
+Start-Process msiexec.exe -ArgumentList "/i node.msi /quiet" -Wait | Wait-Process
+Remove-Item node.msi
+$env:Path = $env:Path + ";C:\Program Files\nodejs"
+npm install -g npm
+npm install -g @angular/cli --loglevel=error
```
Comment (Sand3r-) [inline scripts/InstallNodeDev.ps1]: This line doesn't execute properly on Windows 7 Powershell. It results with following error:
```
Unrecognized token in source text.
At line:1 char:16
+ npm install -g  <<<< @angular/cli --loglevel=error
    + CategoryInfo          : ParserError: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : UnrecognizedToken
```
More: https://github.com/angular/material2/issues/297 and https://stackoverflow.com/questions/38040479/

Creating a batch file for installation should solve the problem

**Comment by Sand3r-** [comment]
Now, the attempt of installation ended up with:
```
The term 'Invoke-WebRequest' is not recognized as the name of a cmdlet, function, script file, or operable program. Che
ck the spelling of the name, or if a path was included, verify that the path is correct and try again.
At C:\Users\Sand3r\Documents\Visual Studio 2015\Projects\spectre\scripts\InstallNodeDev.ps1:3 char:18
+ Invoke-WebRequest <<<<  https://nodejs.org/dist/v6.10.0/node-v6.10.0-x64.msi -OutFile node.msi
    + CategoryInfo          : ObjectNotFound: (Invoke-WebRequest:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

Remove-Item : Cannot find path 'C:\Users\Sand3r\Documents\Visual Studio 2015\Projects\spectre\scripts\node.msi' because
 it does not exist.
At C:\Users\Sand3r\Documents\Visual Studio 2015\Projects\spectre\scripts\InstallNodeDev.ps1:5 char:12
+ Remove-Item <<<<  node.msi
    + CategoryInfo          : ObjectNotFound: (C:\Users\Sand3r...cripts\node.msi:String) [Remove-Item], ItemNotFoundEx
   ception
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.RemoveItemCommand
```
It looks like that Invoke-WebRequest was added in PowerShell version 3.0 or 4.0, while Windows 7 provides version 2.0.

**Comment by gmrukwa** [comment]
[Fix JS solution](https://trello.com/c/H7CLFWPG/62-fix-js-solution)

**Comment by gmrukwa** [comment]
[Remove unnecessary Selenium project](https://trello.com/c/LQvdJUXM/69-remove-unnecessary-selenium-project)

**Pull request merged**